### PR TITLE
feat: #448 Add Timber Sales Manager Field

### PR DESCRIPTION
**Pull request by ianliuwk1019**
API:
	- Add migration for new column project.bcts_manager_name (varchar).
	- Add `bctsMgrName` entity field mapping for column `bcts_manager_name`.
	- Add `bctsMgrName` field for controller's project create/update dto with conditoinal validations.
	- Add validateTimberSalesManager() project service validation for create/update requests.
	- Update openapi swagger spec and recreate api client.

Admin:
	- Add input field for Timber Sales Manager Name with validation and error messages.
	- Display Timber Sales Manager Name at FOM detail and summary page.
	
Public:
	- Display Timber Sales Manager Name at FOM detail.

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
 - [api](https://fom-485.apps.silver.devops.gov.bc.ca/api)
 - [admin](https://fom-485.apps.silver.devops.gov.bc.ca/admin)
 - [public](https://fom-485.apps.silver.devops.gov.bc.ca/public)

Once merged, code will be promoted and handed off to following workflow run.
 - [Main Merge Workflow](https://github.com/bcgov/nr-fom/actions/workflows/merge-main.yml)

### CODE REVIEW

File: api/src/app/modules/project/project.dto.ts [hunk 1]
```diff
@@ -147,6 +152,9 @@ export class ProjectResponse {
 
   @ApiProperty({ required: true })
   operationEndYear: number;
+
```
Comment (basilv) [inline api/src/app/modules/project/project.dto.ts:155]: Shouldn't the bcts name also exist on the create request?

File: api/src/app/modules/project/project.service.ts [hunk 2]
```diff
@@ -169,9 +169,36 @@ export class ProjectService extends DataService<Project, Repository<Project>, Pr
...
+			forestClientName = projectEntity.forestClient.name;
+    }
+    else {
+			const fcl = await this.forestClientService.find([forestClientNumber]);
+			if (fcl && fcl.length == 1) {
+					forestClientName = fcl[0].name;
+			}
+    }
+
+    // Timber Manager's name is required when the holder (foreset client name) starts with "TIMBER SALES MANAGER"
```
Comment (basilv) [inline api/src/app/modules/project/project.service.ts]: Comment is 'wrong' - it isn't 'starts with' but 'contains'

**Review by basilv — COMMENTED** [review]
A couple of minor things.

File: api/src/app/modules/project/project.service.ts (hunk 2 above)
Comment (ianliuwk1019) [reply api/src/app/modules/project/project.service.ts]: Thank you Basil! I made the correction.

File: api/src/app/modules/project/project.dto.ts (hunk 1 above)
Comment (ianliuwk1019) [reply api/src/app/modules/project/project.dto.ts:155]: Hi Basil, correct, the name does exist on the create request on line 59 (`ProjectCreateRequest`).
The line 157 is for the `ProjectResponse`.

**Review by basilv — APPROVED** [review]

**Pull request merged**